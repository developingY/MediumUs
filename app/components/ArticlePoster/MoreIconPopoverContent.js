import styled from 'styled-components';

const MoreIconPopoverContent = styled.button`
  display: inline-block;
  margin: 13px 15px;
  width: 250px;
  line-height: 1.4;
  color: rgba(0, 0, 0, 0.6);
  border: none;
  text-align: left;
  :focus {
    outline: none;
  }
`;

export default MoreIconPopoverContent;
