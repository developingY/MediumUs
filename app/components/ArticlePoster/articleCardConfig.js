export default function getConfig(variation) {
  switch (variation) {
    case 'HomeHeroLeft':
    case 'HomeHeroMid':
    case 'HomeHeroRight':
    case 'HomeList':
    case 'TopicHomepageList':
      return { authorCardVariation: 'Home' };
    case 'TopicHomepageHero': {
      return { authorCardVariation: 'TopicHome' };
    }
    case 'CollectionHomepageHero':
    case 'CollectionHomepageListX2':
    case 'CollectionHomepageListX3':
      return {
        authorCardVariation: 'CollectionHome',
      };
    case 'ArticlePageTitle': {
      return { authorCardVariation: 'ArticleTitle' };
    }
    case 'ArticlePageRecommendation': {
      return { authorCardVariation: 'TopicHome' };
    }
    default:
      return null;
  }
}

// case 'Sidebar': {
//   // different 3 cases, will separate as different component
//   return { authorCardVariation: 'Home' };
// }
