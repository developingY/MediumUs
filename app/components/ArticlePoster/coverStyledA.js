import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  max-width: 680px;
  overflow: hidden;
  white-space: nowrap;
`;

export const Cover = styled.a`
  background-image: url(${props => props.articleCover});
  background-size: cover;
  background-origin: border-box;
  width: 152px;
  @media screen and (max-width: 855px) {
    width: 140px;
  }
  background-position: 50% 50%;
  flex: 0 0 auto;
`;

export const InfoWrapper = styled.div`
  min-width: 0;
  margin-right: 24px;
  // ::before {
  //   content: 'BASED ON YOUR READING HISTORY';
  //   color: rgba(0, 0, 0, 0.54);
  //   letter-spacing: 0.03em;
  //   font-size: 15px;
  //   text-overflow: ellipsis;
  // }
`;
export const TitleWrapper = styled.div`
  margin-bottom: 16px;
`;
export const AuthorWrapper = styled.div``;
