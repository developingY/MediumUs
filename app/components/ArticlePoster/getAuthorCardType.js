export default function getAuthorCardType(variation) {
  switch (variation) {
    case 'HomeHeroLeft':
    case 'HomeHeroMid':
    case 'HomeHeroRight':
    case 'HomeList':
    case 'TopicHomepageList':
      return { authorCardVariation: 'Home' };
    case 'TopicHomepageHero': {
      return { authorCardVariation: 'TopicHome' };
    }
    case 'PublicationHomepageHero':
    case 'PublicationHomepageListX2':
    case 'PublicationHomepageListX3':
      return {
        authorCardVariation: 'PublicationHome',
      };
    case 'ArticlePageTitle': {
      return { authorCardVariation: 'ArticleTitle' };
    }
    case 'ArticlePageRecommendation': {
      return { authorCardVariation: 'TopicHome' };
    }
    default:
      return null;
  }
}

// case 'Sidebar': {
//   // different 3 cases, will separate as different component
//   return { authorCardVariation: 'Home' };
// }
