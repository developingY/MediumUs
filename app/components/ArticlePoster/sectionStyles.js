import { css } from 'styled-components';

export const HomeHeroLeft = {
  coverStyle: css`
    display: block;
    height: 150px;
  `,
};

export const HomeHeroRight = HomeHeroLeft;

export const HomeHeroMid = {
  wrapperStyle: css`
    display: flex;
    @media screen and (max-width: 680px) {
      flex-direction: row-reverse;
      justify-content: space-between;
    }
  `,
  coverStyle: css`
    flex: none;
    height: 100px;
    width: 100px;
    @media screen and (max-width: 680px) {
      width: 140px;
      height: auto;
    }
  `,
  infoWrapperStyle: css`
    margin-left: 24px;
    @media screen and (max-width: 680px) {
      min-width: 0;
      margin-right: 24px;
      margin-left: 0;
    }
  `,
};

export const HomeList = {
  wrapperStyle: css`
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
  `,
  coverStyle: css`
    width: 152px;
    @media screen and (max-width: 855px) {
      width: 140px;
    }
    flex: none;
  `,
  infoWrapperStyle: css`
    min-width: 0;
    margin-right: 24px;
    overflow: hidden;
    text-overflow: ellipsis;
    &:before {
      font-size: 15px;
      white-space: nowrap;
      letter-spacing: 0.03em;
      color: rgba(0, 0, 0, 0.54);
      content: 'POPULAR ON MEDIUM';
      // BASED ON YOUR READING HISTORY/POPULAR ON MEDIUM
    }
  `,
};

export const TopicHomepageHero = {
  coverStyle: css`
    display: block;
    height: 382px;
    @media screen and (max-width: 728px) {
      height: 320px;
    }
  `,
};

export const TopicHomepageList = {
  wrapperStyle: css`
    ${HomeList.wrapperStyle}
  `,
  coverStyle: css`
    width: 152px;
    @media screen and (max-width: 855px) {
      width: 77px;
      height: 77px;
    }
    flex: none;
  `,
  infoWrapperStyle: css`
    ${HomeList.infoWrapperStyle}
  `,
};

export const CollectionHomepageListX2 = {
  coverStyle: css`
    display: block;
    height: 272px;
    border: 1px solid rgba(0, 0, 0, 0.15);
    @media screen and (max-width: 767px) {
      height: 170px;
    }
  `,
};

export const CollectionHomepageListX3 = {
  coverStyle: css`
    ${CollectionHomepageListX2.coverStyle}
    height: 172px;
  `,
};

export const CollectionHomepageHero = {
  wrapperStyle: css`
    display: flex;
    @media screen and (max-width: 767px) {
      display: block;
    }
  `,
  coverStyle: css`
    height 350px;
    flex: 2;
    @media screen and (max-width: 767px) {
      display: block;
      height: 170px;
    }
  `,
  infoWrapperStyle: css`
    flex: 1;
    margin-left: 24px;
    @media screen and (max-width: 767px) {
      margin-left: 0px;
    }
  `,
};

export const ArticlePageTitle = {};

const beforeContent = css`   
        &:before {
          font-size: 16px;
          color: rgba(0, 0, 0, 0.54);
          margin-bottom: 12px;   
          white-space: nowrap; 
          overflow: hidden;
          text-overflow: ellipsis;
          content: 'More from ${props => props.collection || props.name}';
        };`;

export const ArticlePageRecommendation = {
  wrapperStyle: css`
    display: flex;
    flex-direction: column;
    ${beforeContent}
    @media screen and (max-width: 904px) and (min-width: 552px) {
      flex-direction: row-reverse;
      &:before {
        display: none;
      }
    }
  `,
  coverStyle: css`
    padding-bottom: 66.4634%;
    margin-bottom: 16px;
    @media screen and (max-width: 904px) and (min-width: 552px) {
      padding-bottom: 0;
      margin-bottom: 0;
      flex-basis: 50%;
    }
  `,
  infoWrapperStyle: css`
    @media screen and (max-width: 904px) and (min-width: 552px) {
      flex-basis: 50%;
      ${beforeContent}
    }
    overflow: hidden;
  `,
};
