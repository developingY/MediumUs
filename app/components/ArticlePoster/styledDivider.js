import styled from 'styled-components';

export default styled.div`
  width: 1px;
  border-right: 1px solid rgba(0, 0, 0, 0.1);
  margin-left: 12px;
  margin-right: 10px;
  height: 20px;
`;
