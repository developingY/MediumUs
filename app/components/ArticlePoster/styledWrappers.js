import styled, { css } from 'styled-components';
import * as wrapperStyle from './variationWrapperStyles';

const coverBaseStyle = css`
  background-image: url(${props => props.cover});
  background-position: ${props =>
    props.focusPosition ? `${props.focusPosition.join('% ')}%` : `50% 50%`};
  background-size: cover;
  background-origin: border-box;
`;

export const Wrapper = styled.div`
  ${props => wrapperStyle[props.variation].wrapperStyle || ``}
`;
export const Cover = styled.a`
  ${coverBaseStyle};
  ${props => wrapperStyle[props.variation].coverStyle || ``}
`;
export const InfoWrapper = styled.div`
  ${props => wrapperStyle[props.variation].infoWrapperStyle || ``}
`;
export const AuthorWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  ${props => wrapperStyle[props.variation].authorWrapperStyle || ``};
`;

export const IconWrapper = styled.div`
  align-items: center;
  white-space: nowrap;
  //changed
  display: flex;
  //
  ${props => wrapperStyle[props.variation].iconWrapperStyle || ``}
`;

export const ClapText = styled.span`
  margin-left: 5px;
  margin-top: 5px;
  color: rgba(0, 0, 0, 0.54);
`;
