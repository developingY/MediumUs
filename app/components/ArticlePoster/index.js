/**
 *
 * ArticleCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import AuthorCard from '../AuthorCard';
import ArticleTitle from '../ArticleTitle';
import * as wrapperStyles from './variationWrapperStyles';
import {
  Wrapper,
  Cover,
  InfoWrapper,
  AuthorWrapper,
  IconWrapper,
  ClapText,
} from './styledWrappers';
import Divider from './styledDivider';
import getAuthorCardType from './getAuthorCardType';
import { IconButton } from '../Button';
import roundToThousand from '../../utils/roundToThousand';
import OverlayTrigger from '../OverlayTrigger';
import MoreIconPopoverContent from './MoreIconPopoverContent';
import generatePropsValidator from '../../utils/generatePropsValidator';

function ArticlePoster(props) {
  const { authorCardVariation } = getAuthorCardType(props.variation);
  function BookmarkButton() {
    return props.articleInfo.bookmarked ? (
      <IconButton
        type="bookmarkFilledIcon"
        colorSet="pureBlack"
        onClick={props.onRemoveBookmark}
      />
    ) : (
      <IconButton
        type="bookmarkIcon"
        colorSet="black"
        onClick={props.onAddBookmark}
      />
    );
  }

  function renderIcons() {
    switch (props.variation) {
      case 'HomeHeroLeft':
      case 'HomeHeroMid':
      case 'TopicHomepageList':
        return [<BookmarkButton key={props.variation} />];
      case 'HomeList':
        return [
          <BookmarkButton key={props.variation} />,
          <OverlayTrigger
            trigger="click"
            placement="dropdown"
            popoverContent={
              <MoreIconPopoverContent>
                Dismiss this story
              </MoreIconPopoverContent>
            }
            key="moreIcon"
          >
            <IconButton type="moreIcon" colorSet="black" />
          </OverlayTrigger>,
        ];
      case 'ArticlePageTitle':
        return [
          props.articleInfo.twitter && (
            <a href={props.articleInfo.twitter} key="twitter">
              <IconButton type="twitterIcon" colorSet="black" />
            </a>
          ),
          props.articleInfo.linkedIn && (
            <a href={props.articleInfo.linkedIn} key="linkedIn">
              <IconButton type="linkedInIcon" colorSet="black" />
            </a>
          ),
          props.articleInfo.facebook && (
            <a href={props.articleInfo.facebook} key="facebook">
              <IconButton type="facebookSquareIcon" colorSet="black" />
            </a>
          ),
          <BookmarkButton key={props.variation} />,
        ];
      case 'ArticlePageRecommendation':
        return [
          <IconButton
            type="clapSmallIcon"
            colorSet="black"
            key="clapSmallIcon"
          />,
          <ClapText key="claps">
            {roundToThousand(props.articleInfo.claps)}
          </ClapText>,
          <Divider key="divider" />,
          <BookmarkButton key={props.variation} />,
        ];
      default:
        return [];
    }
  }

  return (
    <Wrapper
      variation={props.variation}
      publication={props.authorCardInfo.publication}
      name={props.authorCardInfo.authorName}
    >
      <Cover
        href={props.articleInfo.articleLink}
        variation={props.variation}
        aria-label={props.articleInfo.title}
        cover={props.articleInfo.articleCover}
        focusPosition={props.articleInfo.focusPosition}
      />
      <InfoWrapper
        variation={props.variation}
        publication={props.authorCardInfo.publication}
        name={props.authorCardInfo.authorName}
      >
        <ArticleTitle
          articleLink={props.articleInfo.articleLink}
          title={props.articleInfo.title}
          subtitle={props.articleInfo.subtitle}
          variation={props.variation}
        />
        <AuthorWrapper variation={props.variation}>
          <AuthorCard
            {...props.authorCardInfo}
            variation={authorCardVariation}
            hoverEffect={props.hoverEffect}
          />
          <IconWrapper variation={props.variation}>{renderIcons()}</IconWrapper>
        </AuthorWrapper>
      </InfoWrapper>
    </Wrapper>
  );
}

ArticlePoster.propTypes = {
  // same as AuthorCard.propTypes
  authorCardInfo: PropTypes.shape({
    authorLink: PropTypes.string.isRequired,
    authorName: PropTypes.string.isRequired,
    authorFollowers: generatePropsValidator(['hoverEffect'], 'number'),
    authorDescription: generatePropsValidator(['hoverEffect'], 'string'),
    avatarImg: PropTypes.string.isRequired,
    member: PropTypes.bool.isRequired,
    memberJoinedDate: generatePropsValidator(
      ['member', 'hoverEffect'],
      'string',
    ),
    creatDate: PropTypes.string.isRequired,
    lastModified: PropTypes.string,
    readingTime: PropTypes.string.isRequired,
    premium: PropTypes.bool.isRequired,
    publication: PropTypes.string,
    publicationLink: generatePropsValidator(['publication'], 'string'),
    publicationFollowers: generatePropsValidator(
      ['publication', 'hoverEffect'],
      'number',
    ),
    publicationLogo: generatePropsValidator(
      ['publication', 'hoverEffect'],
      'string',
    ),
    publicationDescription: generatePropsValidator(
      ['publication', 'hoverEffect'],
      'string',
    ),
  }).isRequired,
  articleInfo: PropTypes.shape({
    articleLink: PropTypes.string.isRequired,
    articleCover: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    bookmarked: PropTypes.bool,
    twitter: PropTypes.string,
    facebook: PropTypes.string,
    linkedIn: PropTypes.string,
    claps: PropTypes.number,
    focusPosition: PropTypes.array,
  }).isRequired,

  onAddBookmark: PropTypes.func,
  onRemoveBookmark: PropTypes.func,
  variation: PropTypes.oneOf([..._.keys(wrapperStyles)]).isRequired,
  hoverEffect: PropTypes.bool,
};

export default ArticlePoster;
