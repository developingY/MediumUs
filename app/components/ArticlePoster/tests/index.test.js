/**
 *
 * Tests for ArticleCard
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import ArticlePoster from '../index';
import authorInfo from '../../AuthorCard/stories/exampleData';
import articleInfo from '../stories/exampleData';
import kFormatter from '../kFormatter';
import getAuthorCardType from '../getAuthorCardType';

describe('<ArticleCard />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="HomeHeroLeft"
      />,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have title content', () => {
    const { container } = render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="TopicHomepageList"
      />,
    );
    expect(container).toHaveTextContent(articleInfo.title);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="ArticlePageTitle"
      />,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('Expect to have subtitle content', () => {
    const { container } = render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="TopicHomepageHero"
      />,
    );
    expect(container).toHaveTextContent(articleInfo.subtitle);
  });

  it('Expect second div to have autor name', () => {
    const { container } = render(
      <ArticlePoster {...authorInfo} {...articleInfo} variation="HomeList" />,
    );
    expect(container.querySelectorAll('div')[1]).toHaveTextContent(
      authorInfo.name,
    );
  });

  it('Expect to have subtitle content', () => {
    const { container } = render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="PublicationHomepageHero"
      />,
    );
    expect(container).toHaveTextContent(articleInfo.subtitle);
  });

  it('Expect to have content 4.2K clpas', () => {
    const { container } = render(
      <ArticlePoster
        {...authorInfo}
        {...articleInfo}
        variation="ArticlePageRecommendation"
      />,
    );
    expect(container).toHaveTextContent('4.2K');
  });

  it('Expect to return K formatted number', () => {
    const num = kFormatter(3200);
    expect(num).toEqual('3.2K');
    const n = kFormatter(-32);
    expect(n).toEqual('-32');
  });

  it('Expect default value to be null', () => {
    const type = getAuthorCardType('');
    expect(type).toBe(null);
  });

  it('Expect to have position 50% 50%', () => {
    const article = {
      title:
        'Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art',
      subtitle:
        'And one last warning about their stupidly popular little brother',
      articleLink: './',
      articleCover: './',
      bookmarked: true,
      twitter: './',
      facebook: './',
      linkedIn: './',
      claps: 4230,
    };

    const author = {
      authorLink: './',
      name: 'Lisa Armstrong',
      categoryLink: './',
      date: 'Nov 21',
      readingTime: '13 min read',
      avatarImg: './',
      member: true,
      premium: true,
    };
    const { container } = render(
      <ArticlePoster
        {...author}
        {...article}
        variation="ArticlePageRecommendation"
      />,
    );
    expect(container.querySelector('a')).toHaveStyle(
      `background-position: 50% 50%`,
    );
  });
});
