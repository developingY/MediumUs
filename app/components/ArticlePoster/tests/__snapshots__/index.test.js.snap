// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> Should render and match the snapshot 1`] = `
<div
  class="styledWrappers__Wrapper-p14edy-0 czhChT"
  name="Lisa Armstrong"
  title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
>
  <a
    aria-label="[object Object]"
    class="styledWrappers__Cover-p14edy-1 jAuwIe"
    href="./"
    name="Lisa Armstrong"
    title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
  />
  <div
    class="styledWrappers__InfoWrapper-p14edy-2 KYYYq"
    name="Lisa Armstrong"
    title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
  >
    <div
      class="StyledWrapper__TitleWrapper-sc-111jh54-0 bxiuLE"
      name="Lisa Armstrong"
      title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
    >
      <a
        href="./"
      >
        <h1
          class="StyledWrapper__StyledTitle-sc-111jh54-1 gYdKdB"
          name="Lisa Armstrong"
          title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
        >
          Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art
           
        </h1>
      </a>
      <a
        href="./"
      >
        <div
          class="StyledWrapper__StyledSubtitle-sc-111jh54-2 bomnVF"
          name="Lisa Armstrong"
          title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
        >
          And one last warning about their stupidly popular little brother
        </div>
      </a>
    </div>
    <div
      class="styledWrappers__AuthorWrapper-p14edy-3 cVrZcl"
      name="Lisa Armstrong"
      title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
    >
      <div
        class="AuthorCardsWrappers__StyledContainer-sc-1v4o32t-0 dnQuGq"
      >
        <a
          href="./"
        >
          <div>
            <div
              class="avatarStyle__HaloWrapper-sc-1o8l9nu-2 beQAVp"
            >
              <div
                class="avatarStyle__Halo-sc-1o8l9nu-1 jlerqa"
              >
                <svg
                  fill="inherit"
                  height="91"
                  viewBox="0 0 91 91"
                  width="91"
                >
                  <path
                    clip-rule="evenodd"
                    d="M45.5 1.4c-17.14 0-32 9.95-39.25 24.5L5 25.28C12.47 10.28 27.8 0 45.5 0S78.53 10.29 86 25.28l-1.25.62C77.5 11.35 62.65 1.4 45.5 1.4zM6.25 65.1c7.25 14.55 22.1 24.5 39.25 24.5 17.14 0 32-9.95 39.25-24.5l1.25.62C78.53 80.72 63.2 91 45.5 91S12.47 80.71 5 65.72l1.25-.62z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <img
                alt="Go to the profile of Lisa Armstrong"
                class="avatarStyle__AvtStyledImg-sc-1o8l9nu-0 hngbRH"
                src="IMAGE_MOCK"
              />
            </div>
          </div>
        </a>
        <div
          class="AuthorCardsWrappers__StyledTextContainer-sc-1v4o32t-14 cFKdyf"
        >
          <div
            class="AuthorCardsWrappers__UpTextWrapper-sc-1v4o32t-1 hjAOpW"
          >
            <a
              class="AuthorCardsWrappers__StyledA-sc-1v4o32t-16 eMsgoW"
              href="./"
            >
              Lisa Armstrong
            </a>
             in 
            <a
              class="AuthorCardsWrappers__StyledA-sc-1v4o32t-16 eMsgoW"
              href="./"
            >
              OneZero
            </a>
            <span
              style="margin-left: 8px;"
            >
              <button
                class="OutlinedStyleButton-sc-18pjqv6-0 mypDe"
                type="outlined"
              >
                <span>
                  Follow
                </span>
              </button>
            </span>
          </div>
          <div
            class="AuthorCardsWrappers__DownTextWrapper-sc-1v4o32t-15 gpWgVx"
          >
            <time>
              Nov 21
            </time>
            <span
              class="AuthorCardsWrappers__StyledSpanSpecial-sc-1v4o32t-17 gTquLW"
            >
              Â·
            </span>
            <span>
              13 min read
            </span>
            <span
              class="AuthorCardsWrappers__StyledSpanSpecial-sc-1v4o32t-17 gTquLW"
            >
              <svg
                height="15"
                viewBox="0 0 15 15"
                width="15"
              >
                <path
                  d="M7.44 2.32c.03-.1.09-.1.12 0l1.2 3.53a.29.29 0 0 0 .26.2h3.88c.11 0 .13.04.04.1L9.8 8.33a.27.27 0 0 0-.1.29l1.2 3.53c.03.1-.01.13-.1.07l-3.14-2.18a.3.3 0 0 0-.32 0L4.2 12.22c-.1.06-.14.03-.1-.07l1.2-3.53a.27.27 0 0 0-.1-.3L2.06 6.16c-.1-.06-.07-.12.03-.12h3.89a.29.29 0 0 0 .26-.19l1.2-3.52z"
                />
              </svg>
            </span>
          </div>
        </div>
         
      </div>
      <div
        class="styledWrappers__IconWrapper-p14edy-4 gwPqps"
        name="Lisa Armstrong"
        title="Face Filters for Instagram and Snapchat Are the New Frontier of Surrealist Art"
      >
        <a
          href="./"
        >
          <button
            class="IconStyledButton-sc-1u1vzbn-0 eKfjle"
            type="twitterIcon"
          >
            <svg
              fill="inherit"
              height="25"
              width="25"
            >
              <path
                d="M21.725 5.338c-.744.47-1.605.804-2.513 1.006a3.978 3.978 0 0 0-2.942-1.293c-2.22 0-4.02 1.81-4.02 4.02 0 .32.034.63.07.94-3.31-.18-6.27-1.78-8.255-4.23a4.544 4.544 0 0 0-.574 2.01c.04 1.43.74 2.66 1.8 3.38-.63-.01-1.25-.19-1.79-.5v.08c0 1.93 1.38 3.56 3.23 3.95-.34.07-.7.12-1.07.14-.25-.02-.5-.04-.72-.07.49 1.58 1.97 2.74 3.74 2.8a8.49 8.49 0 0 1-5.02 1.72c-.3-.03-.62-.04-.93-.07A11.447 11.447 0 0 0 8.88 21c7.386 0 11.43-6.13 11.414-11.414.015-.21.01-.38 0-.578a7.604 7.604 0 0 0 2.01-2.08 7.27 7.27 0 0 1-2.297.645 3.856 3.856 0 0 0 1.72-2.23"
              />
            </svg>
          </button>
        </a>
        <a
          href="./"
        >
          <button
            class="IconStyledButton-sc-1u1vzbn-0 eKfjle"
            type="linkedInIcon"
          >
            <svg
              fill="inherit"
              height="29"
              viewBox="0 0 29 29"
              width="29"
            >
              <path
                d="M5 6.36C5 5.61 5.63 5 6.4 5h16.2c.77 0 1.4.61 1.4 1.36v16.28c0 .75-.63 1.36-1.4 1.36H6.4c-.77 0-1.4-.6-1.4-1.36V6.36z"
              />
              <path
                clip-rule="evenodd"
                d="M10.76 20.9v-8.57H7.89v8.58h2.87zm-1.44-9.75c1 0 1.63-.65 1.63-1.48-.02-.84-.62-1.48-1.6-1.48-.99 0-1.63.64-1.63 1.48 0 .83.62 1.48 1.59 1.48h.01zM12.35 20.9h2.87v-4.79c0-.25.02-.5.1-.7.2-.5.67-1.04 1.46-1.04 1.04 0 1.46.8 1.46 1.95v4.59h2.87v-4.92c0-2.64-1.42-3.87-3.3-3.87-1.55 0-2.23.86-2.61 1.45h.02v-1.24h-2.87c.04.8 0 8.58 0 8.58z"
                fill="#fff"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </a>
        <a
          href="./"
        >
          <button
            class="IconStyledButton-sc-1u1vzbn-0 eKfjle"
            type="facebookSqureIcon"
          >
            <svg
              fill="inherit"
              height="29"
              width="29"
            >
              <path
                d="M23.2 5H5.8a.8.8 0 0 0-.8.8V23.2c0 .44.35.8.8.8h9.3v-7.13h-2.38V13.9h2.38v-2.38c0-2.45 1.55-3.66 3.74-3.66 1.05 0 1.95.08 2.2.11v2.57h-1.5c-1.2 0-1.48.57-1.48 1.4v1.96h2.97l-.6 2.97h-2.37l.05 7.12h5.1a.8.8 0 0 0 .79-.8V5.8a.8.8 0 0 0-.8-.79"
              />
            </svg>
          </button>
        </a>
        <button
          class="IconStyledButton-sc-1u1vzbn-0 eKfjle"
          type="bookmarkIcon"
        >
          <svg
            fill="inherit"
            height="25"
            viewBox="0 0 25 25"
            width="25"
          >
            <path
              d="M19 6a2 2 0 0 0-2-2H8a2 2 0 0 0-2 2v14.66h.01c.01.1.05.2.12.28a.5.5 0 0 0 .7.03l5.67-4.12 5.66 4.13a.5.5 0 0 0 .71-.03.5.5 0 0 0 .12-.29H19V6zm-6.84 9.97L7 19.64V6a1 1 0 0 1 1-1h9a1 1 0 0 1 1 1v13.64l-5.16-3.67a.49.49 0 0 0-.68 0z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
